### ISSUE TITLE
make.bat created by "sphinx-apidoc -F" always exits due to "if errorlevel 1" on Windows 7 Professional SP1

### ISSUE THREAD

**Issue by Alexander-Landgraf** [sphinx-doc/sphinx#3922]
### Problem
When I try to create HTML documentation by calling "make.bat html", which has been created by "sphinx-apidoc -F", batch processing stops erroneously because the existance of sphinx is checked by calling "python -msphinx" in line 56 and the condition "if errorlevel 1" in line 57. 

On my Windows 7 Professional SP1 64bit system running Python 3.6.1 32bit and Sphinx v1.6.3, "python -msphinx" always terminates with exit status 1. Therefore make.bat always displays "The Sphinx module was not found. etc. etc." and stops at this point. None of the targets - html in my case - can ever be built. ("python -msphinx --version" on the other hand terminates with exit status 0. This is how I manually patch make.bat as a workaround.)

#### Procedure to reproduce the problem
```
mkdir sphinxbug
cd sphinxbug
echo print("Hello, Sphinx!") >> sphinxbug.py
sphinx-apidoc -F -e -M -H sphinxbug -A "just me" -o .\doc\ .
cd doc
.\make.bat html
```

Output:
```
C:\sphinxbug\doc>.\make.bat html

The Sphinx module was not found. Make sure you have Sphinx installed,
then set the SPHINXBUILD environment variable to point to the full
path of the 'sphinx-build' executable. Alternatively you may add the
Sphinx directory to PATH.

If you don't have Sphinx installed, grab it from
http://sphinx-doc.org/

C:\sphinxbug\doc>
```

#### Error logs / results
```
C:\>python
Python 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 17:54:52) [MSC v.1900 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> ^Z

C:\>python -msphinx
Sphinx v1.6.3
Usage: __main__.py [options] sourcedir outdir [filenames...]

Filename arguments:
  without -a and without filenames, write new and changed files.
  with -a, write all files.
  with filenames, write these.

Options:
  -h, --help          show this help message and exit
  --version           show version information and exit

  General options:
    -b BUILDER        builder to use; default is html
    -a                write all files; default is to only write new and
                      changed files
    -E                don't use a saved environment, always read all files
    -d PATH           path for the cached environment and doctree files
                      (default: outdir/.doctrees)
    -j N              build in parallel with N processes where possible

  Build configuration options:
    -c PATH           path where configuration file (conf.py) is located
                      (default: same as sourcedir)
    -C                use no config file at all, only -D options
    -D setting=value  override a setting in configuration file
    -A name=value     pass a value into HTML templates
    -t TAG            define tag: include "only" blocks with TAG
    -n                nit-picky mode, warn about all missing references

  Console output options:
    -v                increase verbosity (can be repeated)
    -q                no output on stdout, just warnings on stderr
    -Q                no output at all, not even warnings
    --color           Do emit colored output (default: auto-detect)
    -N, --no-color    Do not emit colored output (default: auot-detect)
    -w FILE           write warnings (and errors) to given file
    -W                turn warnings into errors
    -T                show full traceback on exception
    -P                run Pdb on exception

For more information, visit <http://sphinx-doc.org/>.

C:\>echo %errorlevel%
1

C:\>python -msphinx --version
Sphinx (sphinx-build) 1.6.3

C:\>echo %errorlevel%
0

C:\>
```

#### Expected results
"make.bat html" should not stop if sphinx is installed, but build the HTML documentation instead.

### Reproducible project / your project
- n.a. see "Procedure to reproduce the problem" above

### Environment info
- OS: Windows 7 Professional SP1 64bit
- Python version:  Python 3.6.1 32bit
- Sphinx version: v1.6.3

**Comment by tk0miya** [comment]
The implementation appeared at 0832aa73ab.
It seems `make.bat.new_t` compares `errorlevel` with `9009`, not `1`.
https://github.com/sphinx-doc/sphinx/commit/0832aa73ab64d8cf5accbe4a5bbeaeedb7b46a52#diff-d9d20652964aa5efcf2a07b183b7a433L17

I think rewriting this condition resolve this problem. But I don't have windows platform. Could you try this please?

**Comment by Alexander-Landgraf** [comment]
Thanks for looking into this! I confirm that changing the condition from ``if errorlevel 1 (`` to ``if errorlevel 9009 (`` indeed makes it possible to build the documentation.

But as far as I understand the code we also want to detect the situation that sphinx isn't installed. We can't do that via ``errorlevel 9009``:

1. ``errorlevel 9009`` checks for a missing command - **python** in this case.
```
C:\>idontexist
Der Befehl "idontexist" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.

C:\>echo %errorlevel%
9009
```
2. It seems that we have to check for ``errorlevel 1`` to detect a missing module. 
```
C:\>python -midontexist
C:\Python36-32\python.exe: No module named idontexist

C:\>echo %errorlevel%
1
```
3. Now, the problem is, that ``python -m sphinx`` terminates with exit status 1, too. To resolve this indistinguishable situation, I am using ``python -msphinx --version`` in my workaround.

To sum up, the suggested fix solves the reported defect, but doesn't stop batch processing if the sphinx module isn't installed.

**Comment by tk0miya** [comment]
This was fixed at a5f6e97ac45726dcad9dd267279b04d9539e7793.
